dozer-sql/src/pipeline/expression/comparison.rs

```diff
@@ -311,10 +424,104 @@ macro_rules! define_comparison {
+                    Field::Decimal(right_v) => {
+                        let left_val = Decimal::from_str(left_v).map_err(|_| {
+                            PipelineError::UnableToCast(
+                                format!("{}", right_v),
+                                "Decimal".to_string(),
+                            )
+                        })?;
+                        Ok(Field::Boolean($function(left_val, right_v)))
+                    }
+                    Field::Timestamp(right_v) => {
+                        let left_val = DateTime::<FixedOffset>::from_str(left_v).map_err(|_| {
+                            PipelineError::UnableToCast(
+                                format!("{}", right_v),
+                                "Timestamp".to_string(),
+                            )
+                        })?;
+                        Ok(Field::Boolean($function(left_val, right_v)))
+                    }
+                    Field::Date(right_v) => {
+                        let left_val = NaiveDate::from_str(left_v).map_err(|_| {
+                            PipelineError::UnableToCast(format!("{}", right_v), "Date".to_string())
+                        })?;
+                        Ok(Field::Boolean($function(left_val, right_v)))
+                    }
+                    Field::Point(right_v) => {
+                        let left_val = DozerPoint::from_str(left_v).map_err(|_| {
+                            PipelineError::UnableToCast(format!("{}", right_v), "Point".to_string())
+                        })?;
+                        Ok(Field::Boolean($function(left_val, right_v)))
+                    }
+                    Field::Duration(right_v) => {
+                        let left_val = DozerDuration::from_str(left_v).map_err(|_| {
+                            PipelineError::UnableToCast(
+                                format!("{}", right_v),
+                                "Duration".to_string(),
+                            )
+                        })?;
+                        Ok(Field::Boolean($function(left_val, right_v)))
+                    }
+                    Field::Binary(_) | Field::Bson(_) => Err(PipelineError::InvalidTypeComparison(
+                        left_p,
+                        right_p,
+                        $op.to_string(),
+                    )),
+                },
+                Field::Timestamp(left_v) => match right_p {
+                    Field::Timestamp(right_v) => Ok(Field::Boolean($function(left_v, right_v))),
+                    Field::String(right_v) | Field::Text(right_v) => {
+                        let right_val = DateTime::<FixedOffset>::from_str(right_v.as_str())
+                            .map_err(|_| {
+                                PipelineError::UnableToCast(
+                                    format!("{}", right_v),
+                                    "Timestamp".to_string(),
+                                )
+                            })?;
```

Comment:
I think casting this automatically is a bit dangerous, we should keep the date format in consideration.
Probably an explicit cast is better in this case.

Reply:
fixed